### Inferences About the Difference Between Two Population Means: σ1 and σ2 Known

During the 2003 season, Major League Baseball took steps to speed up the play of baseball games in order to maintain fan interest (CNN Headline News, September 30, 2003).
The following results come from a sample of 60 games played during the summer of 2002 and a sample of 50 games played during the summer of 2003. The sample mean shows the
mean duration of the games included in each sample.

      2002 Season                             2003 Season
      n1 = 60                                 n2 = 50 
      Sample mean x1 = 172 min                Sample mean x2 = 166 min

a. A research hypothesis was that the steps taken during the 2003 season would reduce
the population mean duration of baseball games. Formulate the null and alternative
hypotheses.

b. What is the point estimate of the reduction in the mean duration of games during the
2003 season?

c. Historical data indicate a population standard deviation of 12 minutes is a reasonable
assumption for both years. Conduct the hypothesis test and report the p-value. At a .05
level of significance, what is your conclusion?

d. Provide a 95% confidence interval estimate of the reduction in the mean duration of
games during the 2003 season.

    Ans (a) Null Hypothesis, Ho = mu1-mu2 = 0, mu1 = mu2, The steps taken during the 2003 season did not reduce the population mean duration of baseball games.
    
            Alternative Hypothesis, Ha = mu1 > mu2, The steps taken during the 2003 season reduced the population mean duration of baseball games.

            So, we need to use Upper Tail Test

In [77]:
# Ans(b) 
#Point Estimate = x1 - x2 
Point_Estimate = 172-166
Point_Estimate

6

    Ans(b) Point estimate of the reduction in the mean duration of games = 6 min

In [78]:
# Import Libraries
import pandas as pd
import numpy as np
import math
from scipy import stats
from scipy.stats import norm

In [79]:
# Define a function to calculate Z and P value
def Z_and_p(x1, x2, Sigma1, Sigma2, n1, n2):
    z = (x1-x2)/(math.sqrt(((Sigma1**2)/n1)+ ((Sigma2**2)/n2)))
    if(z < 0):
        p = stats.norm.cdf(z)
    else:
        p = 1-stats.norm.cdf(z)
    print(z,p)

In [80]:
# Ans(c) Calculate value for Z and p for given value
# x1 = 172 min, x2 = 166 min, Sigma1 = Sigma2 = 12 min, n1 = 60, n2 = 50
Z_and_p(172,166,12,12,60,50)

2.611164839335468 0.00451171940904016


    Ans(c) Z = 2.61
           p = 0.0045

    Since, p value is less than the Level of Significance (0.05), we reject the null hypothesis and conclude that the steps taken during the 2003 season reduced the population mean duration of baseball games.

In [81]:
# Define a function to calculate Confidence Interval

def Confidence_Interval(x1, x2, sigma, n1, n2, alpha):
   
    # Calculate the standard error of the difference between the means
    std_err = math.sqrt(sigma**2 * (1/n1 + 1/n2))
    
    # Calculate the z-value for the given significance level
    z = norm.ppf(1 - alpha/2)
    
    # Calculate the margin of error
    margin_of_error = z * std_err
    
    # Calculate the lower and upper bounds of the confidence interval
    lower = (x1 - x2) - margin_of_error
    upper = (x1 - x2) + margin_of_error
    
    return (lower, upper)


In [82]:
# Ans(d) Calculate value for Confidence Interval
# x1 = 172 min, x2 = 166 min, Sigma = 12 min, n1 = 60, n2 = 50, alpha = 0.05
Confidence_Interval(172,166,12,60,50,0.05)

(1.4963456423405468, 10.503654357659453)

    Ans(d) Lower Limit = 1.49
           Upper Limit = 10.50

### Inferences About the Difference Between Two Population Means: σ1 and σ2 Unknown and Assume they are equal

Two catalyst are being analyzed to determine how they affect the mean yield of a chemical process. Specifically, catalyst 1 is currently in use, but catalyst 2 is acceptable.
Since catalyst 2 is cheaper, it should be adopted, Providing it does not change the Proceess yield.
A test is run in the plant and results are shown below.
Is there any difference between the mean yields?
Use 0.05, and assume equal variances

    Observation are :
    Catalyst 1 = [91.50, 94.18, 92.18, 95.39, 91.79, 89.07, 94.72, 89.21]
    Catalyst 2 = [89.19, 90.95, 90.46, 93.21, 97.19, 97.04, 91.07, 92.75]

    


Solution:

Hypothesis Test : 
The Pameter of Interest are mu1 and mu2, the mean process yield using catalysts 1 and 2, repectively.

Null Hypothesis = mu1 - mu2 = 0, mu1 = mu2

Alternative Hypothesis = mu != mu2

alpha = 0.05


In [83]:
import scipy.stats as stats

In [84]:
Catalyst_1 = [91.50, 94.18, 92.18, 95.39, 91.79, 89.07, 94.72, 89.21]
Catalyst_2 = [89.19, 90.95, 90.46, 93.21, 97.19, 97.04, 91.07, 92.75]

In [85]:
stats.ttest_ind(Catalyst_1, Catalyst_2, equal_var = True)

Ttest_indResult(statistic=-0.3535908643461798, pvalue=0.7289136186068217)

    t-test = -0.35
    p-value = 0.72
    Conclusion : As p-value is more than alpha value, Null hypothesis cannot be rejected.
    Therefore, we do not have strong evidence to conclude that catalyst 2 results in a mean yield that differs from the mean yield when catalyst 1 is used.
    


### Inferences About the Difference Between Two Population Means: Matched Samples 

Airline travelers often choose which airport to fly from based on flight cost. Cost data (in
dollars) for a sample of flights to eight cities from Dayton, Ohio, and Louisville, Kentucky,
were collected to help determine which of the two airports was more costly to fly from
(The Cincinnati Enquirer, February 19, 2006). A researcher argued that it is significantly
more costly to fly out of Dayton than Louisville. Use the sample data to see whether they
support the researcher’s argument. Use alpha = .05 as the level of significance.

    Hypothesis Test :

    Null Hypothesis : There is no Significant difference in Cost of Dayton and Lousiville flights, Ud, U1-U2 = 0

    Alternative Hypthesis : Dayton flights is costly than Louisville,Ud, U1 - U2 > 0

    U1 = Mean cost for Dayton Flight, U2 = Mean Cost for Louisville Flight 

    It is  Upper Tail Test 


In [86]:
# Import Pandas
import pandas as pd

In [87]:
## Sample data For above Problem
Airline_Data = pd.read_csv('D:\Data for Preprocessing\Airline Data - Updated.csv', index_col= 0)
Airline_Data

Dayton ($)  Lousiville ($)
Destination                                       
Chicago-O’Hare                 319             142
Grand Rapids, Michigan         192             213
Portland, Oregon               503             317
Atlanta                        256             387
Seattle                        339             317
South Bend, Indiana            379             167
Miami                          268             273
Dallas–Ft. Worth               288             274

In [88]:
Dayton_Flight = list(Airline_Data['Dayton ($)'])
Lousiville_Flight = list(Airline_Data['Lousiville ($)'])

In [89]:
t_statistic, p_value = stats.ttest_rel(Dayton_Flight, Lousiville_Flight, alternative = 'greater')
print("t-statistic:", t_statistic)
print("p-value:", p_value)

t-statistic: 1.3206126728426062
p-value: 0.11408375306699065


    As p-value is More than than Alpha (0.05), Therefore we don't reject the Null Hypothesis.

    To conclude this, based on the sample data, we do not have enough evidence to support the researcher's argument 
    that it is significantly more costly to fly out of Dayton than Louisville.

### Inferences About the Difference Between Two Population Proportion

An American Automobile Association (AAA) study investigated the question of whether a
man or a woman was more likely to stop and ask for directions (AAA, January 2006). The
situation referred to in the study stated the following: “If you and your spouse are driving
together and become lost, would you stop and ask for directions?” A sample representative
of the data used by AAA showed 300 of 811 women said that they would stop and ask for
directions, while 255 of 750 men said that they would stop and ask for directions. 

a. The AAA research hypothesis was that women would be more likely to say that they
would stop and ask for directions. Formulate the null and alternative hypotheses for
this study.

b. What is the percentage of women who indicated that they would stop and ask for
directions?

c. What is the percentage of men who indicated that they would stop and ask for
directions?

d. At α = .05, test the hypothesis. What is the p-value, and what conclusion would you
expect AAA to draw from this study? 

    Ans (a) Null hypothesis (H0): There is no difference between the proportion of women and men who would stop and ask for directions. (p_women = p_men)
    
            Alternative hypothesis (Ha): Women are more likely to say that they would stop and ask for directions than men. (p_women > p_men)



In [90]:
# Ans(b)and (c) According to the Data 
# P_Women = 300, P_Men = 255, n_Women = 800, n_Men = 750
Percentage_of_Women = (300/800)*100
Percantage_of_Men = (255/750)*100
print("Percentage of women who would stop and ask for directions:", Percentage_of_Women)
print("Percentage of men who would stop and ask for directions:", Percantage_of_Men)


Percentage of women who would stop and ask for directions: 37.5
Percentage of men who would stop and ask for directions: 34.0


    Ans (b)  Percentage of women who would stop and ask for directions: "37.5"
    Ans (c)  Percentage of men who would stop and ask for directions: "34.0"

In [94]:
# Ans(d)
from scipy.stats import norm

def two_samp_proportion(p1, p2, n1, n2):
    pooled_std_error = np.sqrt((p1 * (1 - p1) / n1) + (p2 * (1 - p2) / n2))
    Z_value = (p1 - p2) / pooled_std_error
    if Z_value < 0:
        p_value = norm.cdf(Z_value)
    else:
        p_value = 1 - norm.cdf(Z_value)
    return Z_value, p_value

# Given data
women_total = 811
women_stop_for_directions = 300

men_total = 750
men_stop_for_directions = 255

# Calculate proportions
p_women = women_stop_for_directions / women_total
p_men = men_stop_for_directions / men_total

# Apply the two-sample proportion test
z_score, p_value = two_samp_proportion(p_women, p_men, women_total, men_total)

print("Z-score:", z_score)
print("P-value:", p_value)

# Checking for significance at alpha = 0.05
alpha = 0.05
if p_value < alpha:
    print("The p-value is less than alpha. AAA would likely reject the null hypothesis.")
    print("There is evidence to suggest that women are more likely to ask for directions.")
else:
    print("The p-value is greater than alpha. AAA would likely fail to reject the null hypothesis.")
    print("There is not enough evidence to suggest a difference in asking for directions between women and men.")


Z-score: 1.235095870918692
P-value: 0.10839741297376726
The p-value is greater than alpha. AAA would likely fail to reject the null hypothesis.
There is not enough evidence to suggest a difference in asking for directions between women and men.


    Ans (d) To conclude, 
            As The p-value is greater than alpha. AAA would likely do not reject the null hypothesis.
            There is not enough evidence to suggest a difference in asking for directions between women and men.